src/utils/memory_size.cpp

```diff
@@ -31,6 +31,9 @@ constexpr auto giga = mega * kilo;
 
 long long in_bytes(const std::string& mem_value)
 {
+    if (mem_value.empty())
+        return 0LL;
```

Comment:
Hmm unsure about that. Shouldn't it be an error? Like… `""` is more `NULL` than `0` :P

Reply:
I just knew this would generate some controversy :wink: I do understand your reluctance here, but I'll try to explain.

I found it necessary to know if memory options were given in the cli when launching a workflow or not.  As can be seen here (lines 2147 & 2148 specifically):
 
https://github.com/canonical/multipass/blob/b3fa1a1452f0001f3284590fbfc0ddf6fd7e7941/src/daemon/daemon.cpp#L2145-L2159

it creates memory entries of 0 bytes which infers that the user did not pass a memory argument.

I don't see any negative side effects when using a `MemorySize` class that is initialized with an empty string to say that it holds 0 bytes as its value.

That said if you (or anyone else on the team) really thinks this is a bad idea, then I can certainly revert this and change the aforementioned lines to look something like:
`MemorySize{request->mem_size().empty() ? "0b" : request->mem_size()};`
which accomplishes the same thing at the expense of some readability.

Reply:
Well, maybe instead we should have an empty `MemorySize`?

Reply:
Ok, that makes sense and will do that instead.

Reply:
Looking at this again a bit more, the `MemorySize` class is not really meant to be "empty" in the sense of not having any value.  Further, the `MemorySize()` constructor (without an argument) defaults to 0 bytes:
https://github.com/canonical/multipass/blob/b3fa1a1452f0001f3284590fbfc0ddf6fd7e7941/src/utils/memory_size.cpp#L68-L70

The functions that are called to get memory sizes in different units all return a `long long` so there is no concept of empty with that unless we specify something like `-1` is considered empty.  So I see these as our options:

- Revert it back to the original where a passed in empty string string is an error and throws an exception and I just do the `MemorySize{request->mem_size().empty() ? "0b" : request->mem_size()};` for my needs.
- Keep it as I have it now in this PR.
- Use `-1` to indicate empty when an empty string is passed in.

I really don't have a strong opinion in any of these options since in the end, they all basically accomplish the same thing for my needs.

Reply:
Ok, will do, but we'll have to decide what an "empty" state is for something that returns `long long` in all of its public functions.

Reply:
This has been reverted back to its original behavior.